Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbNativeWindow.cpp

```diff
@@ -219,17 +231,143 @@ namespace AzFramework
+            handle->offset += readLength;
+        };
+
+        png_structp pngPointer = png_create_read_struct(PNG_LIBPNG_VER_STRING, NULL, NULL, NULL);
+        png_infop pngInfoPointer = png_create_info_struct(pngPointer);
+
+        ReadDataHandle handle = {{ Splash_png, Splash_png_len},0};
+
+        png_set_read_fn(pngPointer, &handle, ReadDataFromInputStream);
+        png_read_info(pngPointer, pngInfoPointer);
+
+        int width = png_get_image_width(pngPointer, pngInfoPointer);
+        int height = png_get_image_height(pngPointer, pngInfoPointer);
+
+        AZStd::unique_ptr<png_bytep[]> rowPointers = AZStd::make_unique<png_bytep[]>(height);
+        //png_bytep rowPointers[height];
+        for (int y = 0; y < height; y++)
+        {
+            rowPointers[y] = (png_byte*)azmalloc(png_get_rowbytes(pngPointer, pngInfoPointer));
+        }
+        png_read_image(pngPointer, rowPointers.get());
+
+        unsigned int stride = width * bytesPerPixel;
+        uint8_t* data = (uint8_t*)malloc(height * stride);  // for parity with free in xcb_image_destroy
+        for (int y = 0; y < height; y++)
+        {
+            for (int x = 0; x < width; x++)
+            {
+                png_bytep px = &(rowPointers[y][x * bytesPerPixel]);
+                int pixelIndex = (x + y * width) * bytesPerPixel; // 3 bytes per pixel (RGB), 1 byte padding
+                data[pixelIndex + 0] = px[2]; // R
+                data[pixelIndex + 1] = px[1]; // G
+                data[pixelIndex + 2] = px[0]; // B
+            }
+            azfree(rowPointers[y]);
+        }
+
+        xcb_void_cookie_t cookie;
+        xcb_pixmap_t pixmap = xcb_generate_id(m_xcbConnection);
+        cookie = xcb_create_pixmap_checked(m_xcbConnection, m_xcbRootScreen->root_depth, pixmap, m_xcbWindow, width, height);
+        if (xcb_generic_error_t* error = xcb_request_check(m_xcbConnection, cookie))
+        {
+            AZ_Printf("XCB window init", "Error in xcb_create_pixmap: %d\n", error->error_code);
+            free(error);  // normal free for malloc/free parity as xcb uses malloc
+            free(data);
+            return;
+        }
+        xcb_image_t* image =
+            xcb_image_create_native(
+                m_xcbConnection,
+                width,
+                height,
+                XCB_IMAGE_FORMAT_Z_PIXMAP,
+                24,
+                data,
+                width * height * bytesPerPixel,
+                data);
+        xcb_image_put(m_xcbConnection, pixmap, m_xcbGraphicContext, image, 0, 0, 0);
+        xcb_image_destroy(image); // also frees data
+        xcb_flush(m_xcbConnection);
+
+        xcb_generic_event_t* event;
+        bool drawn = false;
+        while (!drawn)
+        {
+            event = xcb_wait_for_event(m_xcbConnection);
+            switch (event->response_type & ~0x80)
+            {
+            case XCB_EXPOSE: {
+                xcb_expose_event_t* x = (xcb_expose_event_t*)event;
```

Comment:
nit: Can we use something more descriptive for the `xcb_expose_event_t` type other than `x`. Single variables should be restricted to indexes, loops, but `x` is not descriptive.